### ISSUE TITLE
Text formatting

### ISSUE THREAD

**Issue by tobimensch** [tombh/texttop#51]
Bold, underlined and italic text can add a lot to reading quality (if they are used correctly).

Sub/headings can help the reader to get orientation.

Currently browsh produces pure text both in tty and http-server mode, without any special formattings applied. (The only exceptions being colors)

It will be hard(next to impossible) to display different text sizes in the terminal, but terminals do support bold, underlined and italic text:
https://askubuntu.com/questions/528928/how-to-do-underline-bold-italic-strikethrough-color-background-and-size-i#528938

You can use this to emulate a heading, for example through bold and underlined text.
```echo -e "\e[1;4mbold\e[0m"
```
The http-server can use the usual html tags of course.

To clarify I'm not advocating for emulating all text characteristics of the original source website, I'm just of the opinion that some basic formatting does indeed improve the reading flow and therefore browsh should try to retain a minimal amount of formattings.

**Comment by tombh** [comment]
Yes this is a good point, I've thought of this too. Bold/italic/underline is not consistently supported in terminals, but still we might as well support it. At the moment I actually just increase the brightness of text if it is detected as bold.

Unfortunately the HTTP server can't really use the real HTML tags because it is still fundamentally dependent on monospacing for formatting. It's really and all or nothing thing, as soon as you change the position of one bit of text the whole document has to reflow, at which point you loose any guarantees about the rest of the text reflecting formatting from the original document.
